방장 자동 변경 기능 및 리팩토링

GiHoo opened the pull request:
## 📝작업 내용

> 이번 PR에서 작업한 내용을 간략히 설명해주세요

방장 자동 변경 기능 구현
- 소켓으로 방장이 나갔음을 감지 후 메서드 호출(요청 정보 - roomId, exitHostId)
- 나간 방장의 정보 삭제
- 남은 참여자들의 방 참여 순서(RoomMember 엔티티 createdAt())를 고려하여 그 다음 사람에게 방장 위임

RoomMember chageRole 메서드 역할 분리

## 💬리뷰 시 요구사항 (선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요

작성한 주석처럼 로직을 생각하여 구현해보았습니다~ 
흐름따라 리뷰해주시면 감사하겠습니다!!

GiHoo committed:
feat #74: 방장 자동 변경 기능 구현

- 해당 메서드를 호출하면 요청 정보를 통해 다음을 수행
- 요청 정보의 나간 방장의 번호를 통해 방장 정보 삭제
- RoomMember 엔티티의 createAt()을 참고하여 다음 들어온 사람 탐색
- 방장 역할 위임

GiHoo committed:
mod #74: 요청 정보 및 반환 정보 변경

GiHoo committed:
mod #74: 문서화 정보 추가 및 생성자 정보 추가

GiHoo committed:
feat #74: 방장 자동 변경 정보 반환

GiHoo committed:
feat #74: 방 참여자인지 확인하는 메서드 추가

GiHoo committed:
feat #74: 방장 자동 변경 요청 정보 추가

GiHoo committed:
docs #74: 방장 자동 변경 정보 문서화

GiHoo committed:
refactor #74: 필요없는 주석 제거

GiHoo committed:
refactor #74: 메서드 순서 변경

GiHoo committed:
refactor #74: 방장 자동 변경 반환 dto 네이밍 변경

GiHoo committed:
refactor #74: 방장 수동 변경 요청 dto 네이밍 변경

GiHoo committed:
refactor #74: 방장 수동 변경 반환 dto 네이밍 변경

GiHoo committed:
mod #74: changeRole 책임 분리

- 방장으로 변경
- 참여자로 변경

GiHoo committed:
mod #74: changeRole 책임 분리에 맞게 기능 변경

hyoguoo commented on src/main/java/ei/algobaroapi/domain/room_member/controller/RoomMemberControllerDocImpl.java:
```diff
@@ -43,14 +44,15 @@ public RoomMemberResponseDto changeReadyStatus(@PathVariable(name = "roomId") Lo
-            @RequestBody HostChangeRequestDto hostChangeRequestDto) {
-        return roomMemberService.changeHostManually(hostChangeRequestDto);
+    public RoomHostManualResponseDto changeHostManually(
+            @RequestBody HostManualChangeRequestDto hostManualChangeRequestDto) {
```
👍

hyoguoo reviewed (approved):
전체적인 흐름을 이해하는데 문제 없이 짜주신 것 같습니다!
실제 소켓 연결하면서 더 논의하거나 피드백 드리면 될 것 같아요~~

GiHoo committed:
mod #74: 방장 삭제 기능 제거

- 해당 메서드는 방장이 나갔을 경우 수행되는 메서드
- 나갔다는 행위를 나타내는 exitRoomByMemberId 에서 나간 사람에 대해 삭제 기능 수행하기 때문에 기능 제거

GiHoo committed:
mod #74: 방장 자동 변경 요청 dto 방장 번호 dto 제거

GiHoo committed:
mod: member update 기능 트랜잭션 추가
